docs/source/prgpatterns/lockstep.rst

```diff
@@ -5,11 +5,15 @@ Lockstep Programming Model
 
 .. sectionauthor:: René Widera
 
-The *lockstep programming model* separates code that is evaluated by workers (threads) collectively and independently.
-A problem is described by index domains. A index domain is **independent** from data but **can** be mapped to a data domain one to one but also with more complex mappings.
+The *lockstep programming model* structures code that is evaluated collectively and independently by workers (physical threads).
+Actual processing is described by multi-dimensional index domains of *virtual workers* which can even be changed within a kernel.
```

Comment:
The index domain is always one dimensional `... by one-dimensional index domains ...`